astropy/convolution/core.py

```diff
@@ -88,42 +85,66 @@ def center(self):
         """
         return [axes_size // 2 for axes_size in self._array.shape]
 
+    def calculate_normalization(self, mode='integral'):
+        """
+        Calculate the kernel normalization factor.
+
+        Parameters
+        ----------
+        mode : {'integral', 'peak'}
+            One of the following modes:
+                * 'integral' (default)
+                    Kernel is normalized such that its integral = 1.
+                * 'peak'
+                    Kernel is normalized such that its peak = 1.
+        """
+
+        if mode == 'integral':
+            if self._array.sum() == 0:
+                self._normalization = np.inf
+            else:
+                self._normalization = 1. / self._array.sum()
+        elif mode == 'peak':
+            self._normalization = 1. / self._array.max()
+        else:
+            raise ValueError("invalid mode, must be 'integral' or 'peak'")
+        return self._normalization
+
```

Comment:
Thanks.  It wasn't at all clear to me that `self._normalization` should *always* be `1. / self._array_sum()`.  Actually, I don't think `self._normalization` is really needed at all, except for defining the `_truncation` in the `Gaussian` kernels.  The user should always call `self.normalize()` to normalize the kernel.

`astropy.convolution.convolve` actually doesn't *require* a normalized kernel, only if you want to conserve flux, in which case one should always call `self.normalize()` before `convolve` (not use `self._normalization`).

One problem with setting `self._normalization` to 1 after normalization (instead of recalculating it), is that if one reassigns `kernel.array` then `self._normalization` will be wrong.